### PR TITLE
Fcrepo-2308 - Update Boxc to use fcrepo 5.0.1

### PR DESCRIPTION
**Pull request by bbpennel**

### CODE REVIEW

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java [hunk 1]
```diff
@@ -89,12 +90,13 @@ public FileObject addDescription(InputStream sourceMdStream, String sourceProfil
         FileObject fileObj = createFileObject();
 
         BinaryObject orig = fileObj.addOriginalFile(sourceMdStream, null, "text/plain", null, null);
-        repoObjFactory.createProperty(orig.getPid(), Cdr.hasSourceMetadataProfile, sourceProfile);
-        repoObjFactory.createRelationship(orig.getPid(), RDF.type, Cdr.SourceMetadata);
-        repoObjFactory.createRelationship(pid, PcdmModels.hasRelatedObject, fileObj.getResource());
+        URI metadataUri = RepositoryPaths.getMetadataUri(orig.getPid());
```
Comment (lfarrell) [inline fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java]: Is this being used?

File: services-camel/src/main/java/edu/unc/lib/dl/services/camel/replication/ReplicationProcessor.java [hunk 2]
```diff
@@ -47,6 +47,7 @@
  * @author lfarrell
  *
  */
+@Deprecated
```
Comment (lfarrell) [inline services-camel/src/main/java/edu/unc/lib/dl/services/camel/replication/ReplicationProcessor.java]: How is replication going to work now?

Comment (lfarrell) [reply services-camel/src/main/java/edu/unc/lib/dl/services/camel/replication/ReplicationProcessor.java]: Ah right, longleaf :)

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java (hunk 1 above)
Comment (bbpennel) [reply fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java]: Nope, good catch, I think that was left over from an earlier iteration. Might make same changes to the way descriptions are stored anyways after talking to anna.

File: services-camel/src/main/java/edu/unc/lib/dl/services/camel/replication/ReplicationProcessor.java (hunk 2 above)
Comment (bbpennel) [reply services-camel/src/main/java/edu/unc/lib/dl/services/camel/replication/ReplicationProcessor.java]: Yeah, going to have to add that integration now that the fun of doing it for hyc is complete.

File: fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java (hunk 1 above)
Comment (lfarrell) [reply fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/ContentObject.java]: Were the added changes going to be after this is merged?

**Pull request merged**